src/main/java/iudx/catalogue/server/authenticator/KcAuthenticationServiceImpl.java

```diff
@@ -120,6 +117,15 @@ Future<Boolean> isValidEndpoint(String endpoint) {
     return promise.future();
   }
 
+  private Future<Boolean> isValidAdmin(JwtData jwtData) {
```

Comment:
The duplicate method. `isValidAdmin` method is also there in `JwtAuthServiceImpl`, try to use the same method and avoid code duplication.